hs_rest_api/resources/file_metadata.py

```diff
@@ -65,39 +66,49 @@ def get(self, request, pk, pathname):
                 "start": "2018-02-22",
                 "end": "2018-02-24"
             },
-            "title": "File Metadata Title"
+            "title": "File Metadata Title",
+            "logical_file": {}
         }
         ```
         """
         try:
             resource_file = hydroshare.get_resource_file(pk, pathname)
+            logical_file = resource_file.logical_file
+            metadata = resource_file.metadata
         except ObjectDoesNotExist:
             # Backwards compatibility for file_id
-            resource_file = ResourceFile.objects.get(id=pathname)
-        if resource_file is None:
-            raise NotFound("File {} in resource {} does not exist".format(pathname, pk))
-
-        if resource_file.metadata is None or not resource_file.has_logical_file:
-            raise NotFound("File {} in resource {} has no metadata".format(pathname, pk))
-
-        title = resource_file.metadata.logical_file.dataset_name \
-            if resource_file.metadata.logical_file else ""
-        keywords = resource_file.metadata.keywords \
-            if resource_file.metadata else []
-        spatial_coverage = resource_file.metadata.spatial_coverage.value \
-            if resource_file.metadata.spatial_coverage else {}
-        extra_metadata = resource_file.metadata.extra_metadata \
-            if resource_file.metadata else {}
-        temporal_coverage = resource_file.metadata.temporal_coverage.value if \
-            resource_file.metadata.temporal_coverage else {}
+            try:
+                resource_file = ResourceFile.objects.get(id=pathname)
+                logical_file = resource_file.logical_file
+                metadata = resource_file.metadata
+            except Exception:
+                # is it a folder?
+                resource = hydroshare.get_resource_by_shortkey(pk)
+                dir_path = pk + "/data/contents/"+pathname
+                logical_file = resource.get_folder_aggregation_object(dir_path)
```

Comment:
Good call on the os.path.join

if get_folder_agg it will return None which produces an empty list in the Response, which is consistent with the original implementation.  It'd be better to produce a does not exist return, but I don't want to take that on at the moment.